subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/SourceIncrementalJavaCompilationIntegrationTest.groovy

```diff
@@ -569,6 +569,25 @@ sourceSets {
         output.contains("Full recompilation is required because the source roots could not be inferred.")
     }
 
+    def "missing files are ignored of source roots"() {
+        buildFile << "compileJava.source([file('extra-java'), file('other'), fileTree('missing-tree'), file('missing-file')])"
```

Comment:
Let's break this into multiple lines.

Also, what is the difference between `other` and `missing-file`? Seems to be both missing. For simplicity's sake we could also put the sources under `src/main/java` so we could remove `extra-java` as well.

Reply:
I now only add `missing-tree` and `missing-file`.